lib/recipe/widgets/recipe_ingredients_screen.dart

```diff
@@ -105,93 +123,77 @@ class Ingredients extends StatelessWidget {
-                                .recipes[recipeIndex].ingredients[index].name ??
-                            ''),
-                    subtitle: Text(state.recipes[recipeIndex].ingredients[index]
-                            .description ??
-                        ''),
-                    leading: const Icon(Icons.food_bank),
-                    trailing:
-                        state.recipes[recipeIndex].ingredients[index].saved
-                            ? const Icon(Icons.done)
-                            : const Icon(Icons.rotate_left_rounded),
-                    onTap: () => {
-                      Navigator.pushNamed(context, PageRouter.ingredient,
-                          arguments: {
-                            'recipeIndex': recipeIndex,
-                            'ingredientIndex': index,
-                          })
-                    },
-                  )),
-                );
+                                          .ingredients[index]);
+                                })));
               }));
     });
   }
 }
 
-class DeleteIngredientDialog extends StatelessWidget {
+class IngredientOptions extends StatelessWidget {
   final Recipe recipe;
   final Ingredient ingredient;
 
-  const DeleteIngredientDialog(
+  const IngredientOptions(
       {Key? key, required this.recipe, required this.ingredient})
       : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    String? ingredientName = ingredient.name;
     return BlocBuilder<RecipeBloc, RecipeState>(builder: (context, state) {
-      return AlertDialog(
-        title: const Text('Delete ingredient'),
-        content:
-            Text('Would you like to delete the $ingredientName ingredient?'),
-        actions: [
-          TextButton(
-            onPressed: () => Navigator.pop(context, 'Cancel'),
-            child: const Text('Cancel'),
-          ),
-          TextButton(
-            onPressed: () => {
-              context.read<RecipeBloc>().add(
-                  IngredientDeleted(recipe: recipe, ingredient: ingredient)),
-              Navigator.pop(context, 'Delete')
-            },
-            child: const Text('Delete'),
-          ),
-        ],
-      );
+      final List<Widget> options = [
+        const ListTile(title: Text('Ingredient options')),
+        const Divider(),
+        ListTile(
+          leading: const Icon(Icons.copy),
+          title: const Text('Duplicate'),
+          onTap: () => {
+            context.read<RecipeBloc>().add(
+                IngredientDuplicated(recipe: recipe, ingredient: ingredient)),
+            Navigator.pop(context, 'Duplicate')
+          },
+        ),
+        ListTile(
+          leading: const Icon(Icons.delete),
+          title: const Text('Delete'),
+          onTap: () => {
+            context
+                .read<RecipeBloc>()
+                .add(IngredientDeleted(recipe: recipe, ingredient: ingredient)),
+            Navigator.pop(context, 'Delete')
```

Comment:
What exactly are we achieving here with the additional argument in `Navigator.pop(context, ...)`? i.e. the `'Duplicate'` and `'Delete'`?

Reply:
Was gonna ask you the same about this actually. I noticed them in bits of your code and had included them in case you had reason to. The extra argument serves as a return value, presumably if you needed information as to what popped the context. If it was unintentional, I'm happy to remove it throughout the codebase? (since I don't think we have a need of accessing this return value as of yet)

Reply:
Yeah, I think we don't need them throughout the app, so we can remove all of them. 😊